Dockerfile

```diff
@@ -1,12 +1,12 @@
-FROM node:20.2-alpine as builder
-RUN apk update && apk add curl yarn python3 build-base gcc wget git --no-cache
-RUN curl -sf https://gobinaries.com/tj/node-prune | sh
+FROM node:lts-bullseye-slim as builder

```

Comment:
Why is this necessary? alpine works in docker just fine, and is going to create smaller images. If it works in docker, it works anywhere.

Reply:
Not sure the size matters when the node JS application crashes so often using alpine. This isn't a _needed_ change, but it's a performance and stability one. Especially with the `dumb-init` addition. I am experiencing less headaches and issues by using this custom docker image than using your alpine built one. I thought contributing back to the community a useful change would be beneficial to everyone. 


As for _creates a smaller image_,  I see the following sizes for the images on disk:

```
dessalines/lemmy-ui            latest          a008bbfd24e1       480MB
```

```
custom/lemmy-ui                0.18.3          a87d192120c9      452MB
```

This one is _smaller_, but again not by much not sure why 30MB matter when the image is already nearly half a GB.

Reply:
You should always try to move from debian/ubuntu to alpine if possible for images, musl has better performance than glibc.

Reply:
> musl has better performance than glibc.

No it does not; it makes a smaller environment but it does not have better performance. musl libc is built for correctness and size over performance. This is especially obvious with the implementation of malloc, and rust, leading to a few memory leak issues seen in this project. It also has known issues with regards to DNS stack and often causes `Unknown host` errors from the container.

Reply:
That's debatable, but I've never seen anyone move away from alpine and musl back to ubuntu / glibc. If there were the issues you're talking about, none of the production servers running lemmy-ui would work at all.

Reply:
I realize this is a closed issue, but if anybody else ends up here at some point, I just want to point out that glibc is a totally valid option for Lemmy - lemm.ee has been on glibc from the beginning.


I personally also have had a few very bad experiences with Alpine, strictly because of musl, so I avoid it whenever I can. But this is of course completely anecdotal.